qiskit/optimization/converters/linear_equality_to_penalty.py

```diff
@@ -58,22 +56,16 @@ def encode(self, op: QuadraticProgram) -> QuadraticProgram:
         """
 
         # create empty QuadraticProgram model
-        self._src = copy.deepcopy(op)  # deep copy
-        self._dst = QuadraticProgram()
+        self._src = copy.deepcopy(problem)  # deep copy
```

Comment:
This obviously does deepcopy 🙂  (you didn't add it but we could still remove this comment)
```suggestion
        self._src = copy.deepcopy(problem) 
```